contracts/SSVNetwork.sol

```diff
@@ -359,71 +359,95 @@ contract SSVNetwork is UUPSUpgradeable, Ownable2StepUpgradeable, ISSVNetwork {
         emit ValidatorRemoved(msg.sender, operatorIds, publicKey, cluster);
     }
 
-    function liquidate(address owner, uint64[] memory operatorIds, Cluster memory cluster) external override {
-        bytes32 hashedCluster = cluster.validateHashedCluster(owner, operatorIds, this);
-        cluster.validateClusterIsNotLiquidated();
+    function liquidate(Liquidation[] calldata liquidations) external override {
+        uint32 aggregateValidatorCount;
+        uint256 balanceLiquidatable;
 
-        uint64 clusterIndex;
-        uint64 burnRate;
-        {
-            uint operatorsLength = operatorIds.length;
-            for (uint i; i < operatorsLength; ) {
-                Operator memory operator = operators[operatorIds[i]];
+        Operator[] memory processedOperators = new Operator[](0);
+        uint64[] memory processedIndexes = new uint64[](0);
+        Liquidation[] memory processedLiquidations = new Liquidation[](liquidations.length);
 
-                if (operator.snapshot.block != 0) {
-                    operator.updateSnapshot();
-                    operator.validatorCount -= cluster.validatorCount;
-                    burnRate += operator.fee;
-                    operators[operatorIds[i]] = operator;
-                }
+        uint64 networkFeeIndex = NetworkLib.currentNetworkFeeIndex(network);
 
-                clusterIndex += operator.snapshot.index;
-                unchecked {
-                    ++i;
-                }
+        for (uint i; i < liquidations.length; ++i) {
```

Comment:
I think we must have some number limit of validators to liquidate due to this for loop.  @lior-blox what do you think is a good number here?